odoo/models.py

```diff
@@ -3542,9 +3542,12 @@ def write(self, vals):
                 # put the values of pure new-style fields into cache, and inverse them
                 for record in self:
                     record._cache.update(record._convert_to_cache(new_vals, update=True))
+                inverse_called = []
                 for key in new_vals:
-                    self._fields[key].determine_inverse(self)
-
+                    inverse = self._fields[key].inverse
+                    if inverse and not inverse in inverse_called:
+                        self._fields[key].determine_inverse(self)
+                        inverse_called.append(inverse)
```

Comment:
I suggest you use a `set` instead of a list.
No need to test for `inverse` as, by  definition of `new_vals`, it is always thruthy :-)
```python
inverse_done = set()
for key in new_vals:
    field = self._fields[key]
    if field.inverse not in inverse_done:
        inverse_done.add(field.inverse)
        field.determine_inverse(self)
```